modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/source/FieldSpecificMatcher.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.datastreams.logsdb.qa.matchers.source;
+
+import org.apache.lucene.sandbox.document.HalfFloatPoint;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public interface FieldSpecificMatcher {
+    boolean match(List<Object> actual, List<Object> expected);
```

Comment:
Can this method return a MatchResult instead of a boolean?

Reply:
I wanted to do that but to produce full message in case there is a mismatch you'll need to pass mapping and settings here which ends up being a mouthful.